lib/puppet/util/windows/taskscheduler2.rb

```diff
@@ -0,0 +1,707 @@
+    
+    definition.Settings.Priority
+  end
+
+  # Sets the priority of the task. The +priority+ should be a numeric
+  # priority constant value, from 0 to 10 inclusive
+  #
+  # DONE
+  def priority_value=(value)
+    raise Error.new(_('No currently active task. ITask is NULL.')) if definition.nil?
+    raise TypeError unless value.is_a?(Numeric)
+    raise TypeError if value < 0
+    raise TypeError if value > 10
+
+    definition.Settings.Priority = value
+
+    value
+  end
+
+  def new_task_defintion(task_name)
+    raise Error.new(_("task '%{task}' already exists") % { task: task_name }) if exists?(task_name)
+
+    @pITaskDefinition = @task_service.NewTask(0)
+    @task_defn_path = task_name
+    @pITask = nil
+
+    true
+  end
+
+  # Returns the number of actions associated with the active task.
+  #
+  # DONE
+  def action_count
+    raise Error.new(_('No currently active task. ITask is NULL.')) if definition.nil?
+
+    definition.Actions.count
+  end
+
+  # DONE
+  def action(index)
+    raise Error.new(_('No currently active task. ITask is NULL.')) if definition.nil?
+
+    action = nil
+
+    begin
+      action = definition.Actions.Item(index)
+    rescue WIN32OLERuntimeError => err
+      # E_INVALIDARG 0x80070057 from # https://msdn.microsoft.com/en-us/library/windows/desktop/aa378137%28v=vs.85%29.aspx?f=255&MSPPError=-2147217396
+      if err.message =~ /80070057/m
+        action = nil
+      else
+        raise
+      end
+    end
+    
+    action
+  end
+
+  # DONE
+  def create_action(action_type)
+    raise Error.new(_('No currently active task. ITask is NULL.')) if definition.nil?
+
+    definition.Actions.Create(action_type)
+  end
+
+  # Returns the number of triggers associated with the active task.
+  #
+  # DONE
+  def trigger_count
+    raise Error.new(_('No currently active task. ITask is NULL.')) if definition.nil?
+
+    definition.Triggers.count
+  end
+
+  # Deletes the trigger at the specified index.
+  #
+  # DONE
+  def delete_trigger(index)
+    raise Error.new(_('No currently active task. ITask is NULL.')) if definition.nil?
+
+    definition.Triggers.Remove(index)
+
+    index
+  end
+
+  # Returns a hash that describes the trigger at the given index for the
+  # current task.
+  #
+  # Returns nil if the index does not exist
+  #
+  # Note - This is a 1 based array (not zero)
+  #
+  # DONE
+  def trigger(index)
+    raise Error.new(_('No currently active task. ITask is NULL.')) if definition.nil?
+    
+    trigger = nil
+
+    begin
+      trigger = populate_hash_from_trigger(definition.Triggers.Item(index))
+    rescue WIN32OLERuntimeError => err
+      # E_INVALIDARG 0x80070057 from # https://msdn.microsoft.com/en-us/library/windows/desktop/aa378137%28v=vs.85%29.aspx?f=255&MSPPError=-2147217396
+      if err.message =~ /80070057/m
+        trigger = nil
+      else
+        raise
+      end
+    end
+    
+    trigger
+  end
+
+  # DONE
+  def append_trigger(trigger_hash)
```

Comment:
Do we want to assign a default value here as documentation that this is a hash? Or does a `{}` not really make sense anyhow?